AppController/lib/nginx.rb

```diff
@@ -297,9 +301,14 @@ def self.write_fullproxy_version_config(version_key, http_port, https_port,
     config_path = File.join(SITES_ENABLED_PATH,
                             "appscale-#{version_key}.#{CONFIG_EXTENSION}")
 
-    # Let's reload and overwrite only if something changed.
+    # Let's reload and overwrite only if something changed, or new
+    # certificates have been installed.
     current = ''
     current = File.read(config_path) if File.exists?(config_path)
+
+    # Make sure we have the proper certificates in place.
+    current = '' if ensure_certs_are_in_place(project_id)
```

Comment:
Modifying `current` here just to make it different than `config` seems like a bit of a hack. I would recommend something like
```
certs_changed = ensure_certs_are_in_place(project_id)
if current != config || certs_changed
```